app/templates/ui/app/search/search-results.html

```diff
@@ -1,6 +1,6 @@
 <div ng-repeat="result in results">
   <h4>
-    <a ui-sref="root.view({uri: result.uri})">{{ result.uri }}</a>
+    <a ui-sref="root.view({uri: result.uri})">{{ result.title || result.uri }}</a>
```

Comment:
Another way could be to just loop over response.search.results, and add title/label (I personally like `label` better by the way) before adding it to $scope or ctrl. Such label could be derived from metadata, extracted data, raw content, uri, or something else..

A nice default pattern could be to take the uri, strip away the path, and keep just the doc name as result.label..